ui/app/registration/controllers/patientCommonController.js

```diff
@@ -15,6 +15,57 @@ angular.module('bahmni.registration')
             $scope.genderCodes = Object.keys($rootScope.genderMap);
             $scope.dobMandatory = appService.getAppDescriptor().getConfigValue("dobMandatory") || false;
             $scope.readOnlyExtraIdentifiers = appService.getAppDescriptor().getConfigValue("readOnlyExtraIdentifiers");
+            $scope.showSaveConfirmDialogConfig = appService.getAppDescriptor().getConfigValue("showSaveConfirmDialog");
+            $scope.shouldShowSaveAndContinueButton = false;
+
+            var savedButtonClicked = false;
+
+            var isHref = false;
+
+            $rootScope.onHomeNavigate = function () {
+                if ($scope.showSaveConfirmDialogConfig) {
+                    event.preventDefault();
+                    $scope.targetUrl = event.currentTarget.getAttribute('href');
+                    isHref = true;
+                    $scope.confirmationPrompt(event);
+                }
+            };
+
+            var stateChangeListener = $rootScope.$on("$stateChangeStart", function (event, toState, toParams) {
+                if ($scope.showSaveConfirmDialogConfig) {
+                    $scope.targetUrl = toState.name;
+                    isHref = false;
+                    $scope.confirmationPrompt(event, toState, toParams);
+                }
+            });
+
+            $scope.confirmationPrompt = function (event, toState) {
+                if (savedButtonClicked === false) {
+                    if (event) {
+                        event.preventDefault();
+                    }
+                    ngDialog.openConfirm({template: "../common/ui-helper/views/saveConfirmation.html", scope: $scope});
+                }
+            };
+
+            $scope.continueWithoutSaving = function () {
+                ngDialog.close();
+                savedButtonClicked = true;
```

Comment:
Since `$scope.shouldShowSaveAndContinueButton` is set to `false` in this controller, in which cases does it happen that `savedButtonClicked` is `true`?

Reply:
My bad. Its actually the "Dont Save" button and not the "Save" Button. I'll change the variable name to "dontSaveButtonClicked"